docs/olmv1_overview.md

```diff
@@ -0,0 +1,239 @@
+# OLMv1 Overview
+
+## What won't OLMv1 do that OLMv0 did?
+
+TL;DR: OLMv1 cannot feasibly support multi-tenancy or any feature that assumes multi-tenancy. All multi-tenancy features end up falling over because of the global API system of Kubernetes. While this short conclusion may be unsatisfying, the reasons are complex and intertwined.
+
+Nearly every engineer in the Operator Framework group contributed to design explorations and prototypes over an entire year. For each of these design explorations, there are complex webs of features and assumptions that are necessary to understand the context that ultimately led to a conclusion of infeasibility that led us to today’s conclusion.
+
+Here is a sampling of some of the ideas we explored:
+- [[WIP] OLM v1's approach to multi-tenancy](https://docs.google.com/document/d/1xTu7XadmqD61imJisjnP9A6k38_fiZQ8ThvZSDYszog/edit#heading=h.m19itc78n5rw)
+- [OLMv1 Multi-tenancy Brainstorming](https://docs.google.com/document/d/1ihFuJR9YS_GWW4_p3qjXu3WjvK0NIPIkt0qGixirQO8/edit#heading=h.vy9860qq1j01)
+
+### Watched namespaces cannot be configured in a first-class API
+
+OLMv1 will not have a first-class API for configuring the namespaces that a controller will watch.
+
+Kubernetes APIs are global. Kubernetes is designed with the assumption that a controller WILL reconcile an object no matter where it is in the cluster.
+
+However, Kubernetes does not assume that a controller will be successful when it reconciles an object.
+
+The Kubernetes design assumptions are:
+- CRDs and their controllers are trusted cluster extensions.
+- If an object for an API exists a controller WILL reconcile it, no matter where it is in the cluster.
```

Comment:
If a controller is running with a service account with namespaced RBAC, would the controller even see a CR created in another namespace?

Reply:
No. My understanding is that the design assumption we are stating here is that Kubernetes assumes that there is _a_ controller _somewhere_ that will reconcile an object for an API no matter where it is in the cluster. As far as I am aware this doesn't necessarily mean it _has_ to be a single instance of a controller. I do think in general it makes more sense to use a single controller over many namespace specific controllers though.